Some look & feel updates

apcro opened the pull request:
Most of these are based on personal preference and minor changes that make it easier for me to use the App.

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -57,12 +57,19 @@ class WalletPage extends StatelessWidget {
     );
   }
 
+  var rowCounter = 1;
```
Don't use non-final variables in Stateless Widgets. Instead of mapping the currencies to _buildCurrencyRow like is done in the build method, use List.generate and grab the correct currency using the index. This index can then be used to determine the color of the row.

apcro replied:
I'll look at rewriting this. This is my first time using Flutter so I'm not familiar with List.generate yet.

apcro replied:
I've rewritten this now using List.generate.

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -57,12 +57,19 @@ class WalletPage extends StatelessWidget {
+  var rowCounter = 1;
+  var currencyRowColour = Colors.white;
   Widget _buildCurrencyRow(BuildContext context, Currency currency) {
+    currencyRowColour = (rowCounter % 2 == 0 ? currencyRowColour = Colors.amberAccent.shade100 : currencyRowColour = Colors.transparent);
```
A color like light gray is probably easier on the eyes than amber. Try something like Colors.white10.
Also keep in mind that amber doesn't look right when using the dark theme. Colors.white10 will probably look more in place when using dark theme. Alternatively you can check which theme is selected and determine the color based on that.

apcro replied:
I've re-done this with theme-respecting colours.

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -78,19 +85,9 @@ class WalletPage extends StatelessWidget {
   }
 
   Widget _buildCurrency(BuildContext context, Currency currency) {
-    if (currency.name == 'Coin' || currency.id == 1) {
```
Don't remove the check for the Coin currency. This displays an alternative widget that properly formats Gold/Silver/Copper. For example: removing this check changes having 300 gold from displaying like: 300g, 0s, 0c to 3.000.000 gold.

apcro replied:
I've put this back.

DanielScholte commented on lib/pages/tab.dart:
```diff
@@ -116,6 +116,7 @@ class _TabPageState extends State<TabPage> {
         hasInk: true,
         inkColor: Color.fromRGBO(0, 0, 0, .15),
         onTap: (index) => setState(() => _currentIndex = index),
+        borderRadius: BorderRadius.vertical(top: Radius.circular(12)),
```
A borderRadius is unnecessary when the BottomNavigation widget doesn't have a different background color or shadow. In this case, it only results in an off looking shadow when using light theme on the trading post tab.

apcro replied:
This was a change I was playing with that I missed reverting.

DanielScholte commented on lib/pages/character/inventory.dart:
```diff
@@ -90,42 +90,56 @@ class InventoryPage extends StatelessWidget {
   Widget _buildBag(BuildContext context, Bags bag, int bagIndex) {
```
I'm not a big fan of the changes to the Inventory page. Instead of giving the headers a different color (which is a bit too much imo), maybe add a separator in-between the header and the items?

apcro replied:
I've updated this with a Divider() widget instead of the coloured background, but only when there's something to show in the bag slot.

DanielScholte commented on lib/pages/character/inventory.dart:
```diff
@@ -90,42 +90,56 @@ class InventoryPage extends StatelessWidget {
               width: double.infinity,
-              margin: EdgeInsets.only(top: 8.0),
+              margin: EdgeInsets.only(top: 8.0, bottom: 8.0),
+              alignment: AlignmentDirectional.center,
```
I personally think the left alignment looks better.

apcro replied:
This is very much a personal thing, I've always preferred the centered style.

DanielScholte replied:
Personally I'd rather have you revert this back to left alignment, so it aligns better with the bag icon.
Also change the margin to `EdgeInsets.all(8.0)`.

DanielScholte commented on lib/pages/character/inventory.dart:
```diff
@@ -90,42 +90,56 @@ class InventoryPage extends StatelessWidget {
   Widget _buildBag(BuildContext context, Bags bag, int bagIndex) {
     List<InventoryItem> inventory = bag.inventory.where((i) => i.id != -1 && i.itemInfo != null).toList();
     int usedSlots = inventory.length;
+    FontStyle fontStyle = (usedSlots == bag.size ? FontStyle.italic : FontStyle.normal);
```
What's your reason for changing the font style of bags that are full?

apcro replied:
The widget doesn't currently show empty slots, so this is a quick visual change to display 'full' without having to cognitively parse the numbers on the right.

DanielScholte reviewed (changes requested):
Thanks for the pull request!
Added some comment regarding the changes.  
I'm also not sure what i think about the changes to the border radius. I'm going to ask around what people prefer and I'll come back to you about that.
Could you also verify that lowering the size and quality of the images doesn't result in blurry images on higher resolution devices (1440p and 4k)?

apcro commented:
> Could you also verify that lowering the size and quality of the images doesn't result in blurry images on higher resolution devices (1440p and 4k)?

The images were shrunk using https://tinypng.com/, with the intent solely to save on image size, the dimensions weren't changed however I've reverted the small icons as after compression they appear marginally darker.

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -43,7 +48,7 @@ class WalletPage extends StatelessWidget {
                 },
                 child: CompanionListView(
-                  children: state.currencies.map((c) => _buildCurrencyRow(context, c)).toList(),
+                  children: _currencies.asMap().entries.map((entry) => _buildCurrencyRow(context, entry.key, entry.value)).toList(),
```
Looks better, but what i meant was using the List.generate here to get both the index and the currency. For example:  
```
children: List.generate(state.currencies.length, (index) => _buildCurrencyRow(context, index, state.currencies[index]))
```

You can then also remove the three lines at 39 to 41.

apcro replied:
Ah, that's clever. I'm learning as I'm going :)

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -57,12 +62,19 @@ class WalletPage extends StatelessWidget {
         children: <Widget>[
+          Padding(padding: EdgeInsets.only(left: 8.0)),
           Expanded(
+            
```
Remove the empty line

apcro replied:
Updated

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -57,12 +62,19 @@ class WalletPage extends StatelessWidget {
-  Widget _buildCurrencyRow(BuildContext context, Currency currency) {
+  Widget _buildCurrencyRow(BuildContext context, int index, Currency currency) {
+    Color currencyRowColour = Colors.white12;
+    Color highlightColor = Theme.of(context).brightness ==  Brightness.light ? Colors.black12 : Colors.white12;
+    currencyRowColour = (index % 2 != 0 ? currencyRowColour = highlightColor : currencyRowColour = Colors.transparent);
     return Container(
-      margin: EdgeInsets.all(8.0),
+      margin: EdgeInsets.all(0.0),
+      padding: EdgeInsets.only(left: 0.0, right: 8.0, top: 8.0, bottom: 8.0),
+      color: currencyRowColour,
```
Code can be a bit neater. Change the code at the top to just this:
```
Color highlightColor = Theme.of(context).brightness ==  Brightness.light ? Colors.black12 : Colors.white12;
```
Then when deciding the Container color, do this:
```
color: index % 2 != 0 ? highlightColor : null,
```

apcro replied:
Updated

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -57,12 +62,19 @@ class WalletPage extends StatelessWidget {
     return Container(
-      margin: EdgeInsets.all(8.0),
+      margin: EdgeInsets.all(0.0),
+      padding: EdgeInsets.only(left: 0.0, right: 8.0, top: 8.0, bottom: 8.0),
```
0 values when using EdgeInsets are unnecessary. You could remove those and change it to:
```
padding: EdgeInsets.only(top: 8.0, bottom: 8.0),
```
However, you can make it look cleaner by using symmetric:
```
padding: EdgeInsets.symmetric(vertical: 8.0),
```

DanielScholte replied:
On second look, I'd rather have you make this `padding: EdgeInsets.all(8.0),` as there is no padding on the right side at the moment.

apcro replied:
Updated.

DanielScholte commented on lib/pages/home/wallet/wallet.dart:
```diff
@@ -57,12 +62,19 @@ class WalletPage extends StatelessWidget {
+    currencyRowColour = (index % 2 != 0 ? currencyRowColour = highlightColor : currencyRowColour = Colors.transparent);
     return Container(
-      margin: EdgeInsets.all(8.0),
+      margin: EdgeInsets.all(0.0),
```
This line can be removed because you set the margin to 0, which is the default.

apcro replied:
Updated

DanielScholte commented on lib/pages/character/inventory.dart:
```diff
@@ -90,61 +90,77 @@ class InventoryPage extends StatelessWidget {
-                  .toList(),
+                    .where((i) => i.id != -1)
+                    .map((i) =>
+                    CompanionItemBox(
```
I'd revert this part, as I personally think it looked better previously.

apcro replied:
This isn't one of my changes?

DanielScholte commented on lib/pages/character/inventory.dart:
```diff
@@ -90,61 +90,77 @@ class InventoryPage extends StatelessWidget {
     int usedSlots = inventory.length;
+    FontStyle fontStyle = (usedSlots == bag.size ? FontStyle.italic : FontStyle.normal);
     return CompanionCard(
+      padding: EdgeInsets.all(0.0),
```
This can be changed to `EdgeInsets.zero` which looks a bit neater.

apcro replied:
I've done this.

DanielScholte commented on lib/pages/character/inventory.dart:
```diff
@@ -90,61 +90,77 @@ class InventoryPage extends StatelessWidget {
+            ),
           ),
+          if (usedSlots > 0)
+            Divider(),
```
I've been playing around, and I think you can improve the Divider by setting it's parameters like this:
```
Divider(
    height: 2,
    thickness: 1,
)
```
But feel free to come with other suggestions 😃

apcro replied:
I've updated this to 2/1 - I fiddled with the numbers, these look good. The intent was just to provide a bit of a differentiation between the heading and the content, so this works really well.

DanielScholte reviewed (changes requested):
Changes look good! Added a few more points regarding slight UI changes or code cleanup.
Got a few more requests:
- I've asked the opinion of a few app users, and most people seem to prefer the previous borderRadius of 12.0. Could you revert the changes back to 4 to 12?
- Can you change the target branch from master to development? The changes will then be implemented in version 1.5 of the app once the event notifications are finished.

DanielScholte reviewed (changes requested):
Thanks for the changes :)  
Did you push the commits? As it looks like they are currently missing from the pull request.
Furthermore, could you look at the message of the previous review? It contains two more points that would be great if you could take a look at them.

apcro commented:
> * I've asked the opinion of a few app users, and most people seem to prefer the previous borderRadius of 12.0. Could you revert the changes back to 4 to 12?

I've reverted these changes now.

> * Can you change the target branch from master to development? The changes will then be implemented in version 1.5 of the app once the event notifications are finished.

And I've moved all the changes to a new `development` branch. I'll withdraw this pull request and make a new one on the new branch.
